src/views/Design.vue

```diff
@@ -0,0 +1,363 @@
+import NewOrganism from "@/components/NewOrganism.vue";
+import NewModel from "@/components/NewModel.vue";
+import { NextHandler } from "@/router";
+import { RuleHandler, RuleOutcome } from "@/types/forms";
+
+interface ProductItem {
+  name: string;
+}
+
+interface DesignState {
+  isValid: boolean;
+  project?: ProjectItem;
+  projectRules: ReadonlyArray<RuleHandler>;
+  isProjectCreationDialogVisible: boolean;
+  organism?: OrganismItem;
+  organismRules: ReadonlyArray<RuleHandler>;
+  isOrganismCreationDialogVisible: boolean;
+  product?: ProductItem;
+  productRules: ReadonlyArray<RuleHandler>;
+  productOptions: ProductItem[];
+  isLoadingProducts: boolean;
+  isAerobic: boolean;
+  showAdvanced: boolean;
+  bigg: boolean;
+  rhea: boolean;
+  model?: ModelItem;
+  isModelCreationDialogVisible: boolean;
+  maxPredictions: number;
+  predictionRules: ReadonlyArray<RuleHandler>;
+  isSubmitted: boolean;
+  hasSubmissionError: boolean;
+}
+
+export default Vue.extend({
+  name: "Design",
+  components: {
+    NewProject,
+    NewOrganism,
+    NewModel
+  },
+  data(): DesignState {
+    return {
+      isValid: false,
+      project: undefined,
+      projectRules: [v => !!v || "Project is required"],
+      isProjectCreationDialogVisible: false,
+      organism: undefined,
+      organismRules: [v => !!v || "Organism is required"],
+      isOrganismCreationDialogVisible: false,
+      product: undefined,
+      productRules: [v => !!v || "Product is required"],
+      productOptions: [],
+      isLoadingProducts: true,
+      isAerobic: false,
+      showAdvanced: false,
+      bigg: false,
+      rhea: true,
+      model: undefined,
+      isModelCreationDialogVisible: false,
+      maxPredictions: 3,
+      predictionRules: [
+        (v: number) => (v > 0 && v <= 10) || "Must be between 1 and 10"
+      ],
+      isSubmitted: false,
+      hasSubmissionError: false
+    };
+  },
+  computed: {
+    projectOptions(): ProjectItem[] {
+      return this.$store.state.projects.projects;
+    },
+    // TODO (Moritz Beber): Selection of organism in future should depend on the
+    //  selected project, i.e., you can only select public organisms and
+    //  organisms within that project.
+    // See https://github.com/DD-DeCaF/caffeine-vue/issues/42
+    organismOptions(): OrganismItem[] {
+      return this.$store.state.organisms.organisms;
+    },
+    allModels(): ModelItem[] {
+      return this.$store.state.models.models;
+    },
+    modelOkay() {
+      return !!this.project && !!this.organism;
+    },
+    modelOptions(): ModelItem[] {
+      if (this.modelOkay) {
+        return this.allModels.filter(
+          (model: ModelItem) =>
+            model.organism_id === this.organism.id &&
+            (model.project_id == null || model.project_id === this.project.id)
+        );
+      } else {
+        return [];
+      }
+    },
+    modelRules(): RuleHandler[] {
+      return [this.modelOkayRule, v => !!v || "Model is required"];
```

Comment:
This doesn't quite work as expected. If you select a project, then an organism with no mapping (e.g. cricetulus griseus), then click the model select, then deselect it by clicking somewhere else, you still get the "Please first select a project and organism" error (expected "Model is required").

Reply:
My assumption was that there should always be at least one model for an organism. However, I now see that this may not always be the case.

Reply:
I'm not sure it can be done with Vuetify only. Would have to manually call validation upon the field losing focus.

Reply:
How about having the validation only be concerned with whether the model is set, and instead use a `:hint` if project and/or organism is missing?